src/controller.cpp

```diff
@@ -529,14 +522,13 @@ Controller::UpdateResult Controller::update(
   ROS_DEBUG("d_end_phase: %f", d_end_phase);
   ROS_DEBUG("distance_to_goal: %f", distance_to_goal_);
 
-  // Get 'angle' towards current_goal
-  tf2::Transform base_to_goal = current_tf.inverseTimes(current_goal_);
-  const double angle_to_goal = atan2(base_to_goal.getOrigin().x(), -base_to_goal.getOrigin().y());
+  const auto in_direction_of_goal =
+    !std::signbit(current_tf.getOrigin().dot(current_goal_.getOrigin()));
```

Comment:
I was thinking something along the lines of this:
```suggestion
const auto base_to_next_path_pose = current_goal_.getOrigin() - current_tf.getOrigin();
  const auto in_direction_of_goal =
    !std::signbit(current_goal_.getBasis().tdotx(base_to_next_path_pose)*target_x_vel);
```
But I'm not sure if it's entirely correct. I now project onto the current_goal_ x vector, but one could also pick the current_tf x vector. Not sure which is correct, and how to test this in an integration test.

Also, I was inspired by your `!std::signbit`, and added the multiplication with the current velocity setpoint. That doesn't make it any more transparent...

Reply:
I'm not sure if it's correct or not but it did give different results than the original implementation (for certain inputs). Based on your remarks, I changed it to the following: 

```cpp
  const auto base_to_next_path_pose = current_goal_.getOrigin() - current_tf.getOrigin();
  const auto in_direction_of_goal =
    !std::signbit(current_tf.getBasis().tdotx(base_to_next_path_pose) * target_x_vel);
```

This does give the same results as the original implementation for all inputs that occur in the automated tests.

I plan to do the following:
- move the new implementation to a named function in `calculations.h/cpp`
- add some unit tests

I could use some help with the documentation and coming up with interesting inputs for the unit tests